src/Storages/ObjectStorage/DataLakes/IcebergMetadata.cpp

```diff
@@ -403,23 +401,67 @@ DataLakeMetadataPtr IcebergMetadata::create(
     const Poco::JSON::Object::Ptr & object = json.extract<Poco::JSON::Object::Ptr>();
 
     auto format_version = object->getValue<int>("format-version");
-    auto [schema, schema_id]
-        = parseTableSchema(object, format_version, local_context->getSettingsRef()[Setting::iceberg_engine_ignore_schema_evolution]);
-
-    auto current_snapshot_id = object->getValue<Int64>("current-snapshot-id");
     auto snapshots = object->get("snapshots").extract<Poco::JSON::Array::Ptr>();
 
     String manifest_list_file;
-    for (size_t i = 0; i < snapshots->size(); ++i)
+    Int32 schema_id = -1;
```

Comment:
I assume (maybe naively) that schema id is somehow visible to the users of iceberg, would it be convenient to set it directly, rather than do exhaustive search on metadata JSON?

If so, maybe you should consider adding another setting, that would allow user to specify schema_id directly?

Reply:
It's available in the JSON but not really something Iceberg end users think about.

What we're actually searching for is the `manifest_list_file` for either the current (latest) snapshot, or the one closest to the `iceberg_query_at_timestamp_ms`. And as part of getting that manifest file we also grab the `schema_id` for that snapshot. This just got extracted from [parseTableSchema](https://github.com/ClickHouse/ClickHouse/blob/4bfce22caaa2afdb86ca034e33c94cc6dc454527/src/Storages/ObjectStorage/DataLakes/IcebergMetadata.cpp#L251) in this PR because the code on master is hardcoded to only look for the current snapshot/schema.